core/src/main/java/org/web3j/tx/gas/ContractGasProvider.java

```diff
@@ -20,8 +20,11 @@
     @Deprecated
```

Comment:
I don't understand why this method is `@Deprecated`. There shouldn't be a link between `contractFunc` and `gasPrice`.

This can be modified in this PR also.

Reply:
So what is the desired interface of the ContractGasProvider? I am not familiar with how do other developers implement or use it. In my understanding, the 'getGasPrice' does not need any parameters as it is not something that depends on the kind of transaction you are about to make. 
On the other hand, 'getGasLimit' needs all the parameters, and it can throw the exception here (did not add it to the interface yet, but I am not a big fan of catching it inside the EstimatedGasProvider).

So what do you think?  Should I 'undepricate' the "getGasPrice()" and deprecate the "getGasPrice(String contractFunc)" here?

Reply:
Yes agreed. You can change the deprecated `getGasPrice`.

Regarding the exception handling, this should be handled by the caller. It is ok to remove the `try/catch` and declare that the method `throws` an exception

Reply:
I think it is good that the funcName is in the method signature because:
* it doesn't hurt if you don't use it
* it is important for me as I adjust the gas price for some functions (some need faster handling, some can live with slower execution).